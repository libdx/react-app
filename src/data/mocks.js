//@flow

import type { Movie } from '../movie'

export const forrest: Movie = {
    imageURL: "http://t0.gstatic.com/images?q=tbn:ANd9GcSppDgk99BKVA4TJtWc1FN4-HUkdWrFNfMm1-M0nQ01sIOcbTZu",
    title: "Forrest Gump",
    releaseYear: "1994",
    genre: "Comedy-drama",
    description: "Slow-witted Forrest Gump (Tom Hanks) has never thought of himself as disadvantaged, and thanks to his supportive mother (Sally Field), he leads anything but a restricted life. Whether dominating on the gridiron as a college football star, fighting in Vietnam or captaining a shrimp boat, Forrest inspires people with his childlike optimism. But one person Forrest cares about most may be the most difficult to save -- his childhood love, the sweet but troubled Jenny (Robin Wright)."
}

export const movies = [forrest, forrest, forrest, forrest, forrest, forrest]

