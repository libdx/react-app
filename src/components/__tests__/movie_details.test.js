import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import MovieDetails from '../movie_details'
import MovieTitle from '../movie_title'
import MovieGenreLabel from '../movie_genre_label'
import MoviePoster from '../movie_poster'
import MovieDescription  from '../movie_description'

// FIXME: fire out how to properly mock model objects
import { forrest as movie } from '../../data/mocks'

describe('MovieDetails', () => {

    let element
    let component

    beforeEach(() => {
        element = <MovieDetails movie={movie} />
        component = shallow(element)
    })

    it('has movie title', () => {
        const title = component.find('MovieTitle') 
        expect(title.exists()).toBe(true)
    })

    it('has movie genre label', () => {
        const label = component.find('MovieGenreLabel') 
        expect(label.exists()).toBe(true)
    })

    it('has movie poster', () => {
        const poster = component.find('MoviePoster') 
        expect(poster.exists()).toBe(true)
    })

    it('has movie description', () => {
        const description = component.find('MovieDescription') 
        expect(description.exists()).toBe(true)
    })

    it('renders correctly', () => {
        const tree = renderer.create(element).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
