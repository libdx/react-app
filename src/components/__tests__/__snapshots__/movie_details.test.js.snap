// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<div
  className="card movie-details"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <img
        className="card-img-top poster-img"
        src="https://image.tmdb.org/t/p/w500/yE5d3BUhE8hCnkMUJOo1QDoOGNz.jpg"
      />
    </div>
    <div
      className="col-md-8 card-body movie-details-body"
    >
      <div
        className="movie-details-title-box"
      >
        <h6
          className="card-title textbox"
        >
          <span
            className="alignleft"
          >
            Forrest Gump (8.3)
          </span>
          <span
            className="badge badge-light badge-margin-left"
          >
            1994
          </span>
          <div
            className="alighclear"
          />
        </h6>
      </div>
      <p
        className="card-subtitle mb-2 text-muted"
      >
        Comedy, Drama, Romance
      </p>
      <p>
         
        A man with a low IQ has accomplished great things in his life and been present during significant historic events - in each case, far exceeding what anyone imagined he could do. Yet, despite all the things he has attained, his one true love eludes him. 'Forrest Gump' is the story of a man who rose above his challenges, and who proved that determination, courage, and love are more important than ability.
         
      </p>
    </div>
  </div>
</div>
`;
